lib/components/OverdriveProvider/OverdriveProvider.tsx

```diff
@@ -1,43 +1,134 @@
-import React, { useEffect } from 'react';
+import { invariant } from '@autoguru/utilities';
+import { assignInlineVars } from '@vanilla-extract/dynamic';
+import { isEqual } from 'es-toolkit';
+import React, { createContext, useContext, useEffect, useMemo } from 'react';
 
+import baseTheme from '../../themes/base';
+import { makeRuntimeTokens, type RuntimeTokens } from '../../themes/makeTheme';
+import { themeContractVars } from '../../themes/theme.css';
+import { type BreakPoints } from '../../themes/tokens';
 import { isBrowser } from '../../utils';
-import {
-	ThemeProvider,
-	type ThemeProviderProps,
-} from '../ThemeProvider/ThemeProvider';
-
-export interface OverdriveProviderProps extends ThemeProviderProps {
-	/**
-	 * When set to true theme className must be present in a parent dom element of your components including portals.
-	 * OD Portal component automatically adds theme class to the mounting node
-	 */
+
+import { useColorOverrides, type ColorOverrides } from './useColorOverrides';
+
+type ThemeContract = typeof themeContractVars;
+type PortalMountPoint = React.RefObject<HTMLElement | null>;
+
+export interface ProviderProps {
+	/** The theme object to be used throughout the application */
+	theme?: typeof baseTheme;
+	/** Custom breakpoints for responsive design */
+	breakpoints?: BreakPoints;
```

Comment:
This is an interesting change. How are you asserting that when customized, that the component still behaves/looks as the intended design? I understand there may be an internal constraint that called for this, but still curious.